### PR TITLE
Add ucmerced staging hub with profileList for Python and R images

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Fixes https://github.com/2i2c-org/infrastructure/issues/3150

## Related

- #2674
- #2583
- https://github.com/2i2c-org/infrastructure/issues/3188

### CODE REVIEW

File: config/clusters/2i2c/ucmerced-staging.values.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          - https://accounts.google.com/o/oauth2/auth
+        allowed_idps:
+          urn:mace:incommon:ucmerced.edu:
+            username_derivation:
+              username_claim: "eppn"
+          http://google.com/accounts/o8/id:
+            username_derivation:
+              username_claim: "email"
+            allowed_domains:
+              - "2i2c.org"
```
Comment (consideRatio) [inline config/clusters/2i2c/ucmerced-staging.values.yaml:76]: - relocated `Authenticator` config to the bottom to be consistent in how other config is ordered
- removed `shown_idps` for oauth16
- added `allow_all` for the ucmerced.edu idp for oauth16 equivalent config

```suggestion
      JupyterHub:
        authenticator_class: cilogon
      CILogonOAuthenticator:
        oauth_callback_url: https://staging.ucmerced.2i2c.cloud/hub/oauth_callback
        allowed_idps:
          urn:mace:incommon:ucmerced.edu:
            username_derivation:
              username_claim: "eppn"
          http://google.com/accounts/o8/id:
            username_derivation:
              username_claim: "email"
            allowed_domains:
              - "2i2c.org"
      Authenticator:
        admin_users:
          - schadalapaka@ucmerced.edu
```

Related to #3144 being merged or not, and I'm looking to merge it in 30 minutes or so after reviewing all config once more

**Review by sgibson91 — COMMENTED** [review]
Is it worth having a `ucmerced-common.values.yaml` file to reduce repetition, e.g., homepage config?

File: config/clusters/2i2c/ucmerced-staging.values.yaml (hunk 1 above)
Comment (GeorgianaElena) [reply config/clusters/2i2c/ucmerced-staging.values.yaml]: Thanks @consideRatio! I'll wait for the merge (🚀 🚀 🚀 ) to rebase this and manually deploy to double check config.

**Comment by GeorgianaElena** [comment]
Thanks @consideRatio and @sgibson91 for the review. I've updated the PR based on the feedback. Also, I've manually deployed this and is runnitng at https://staging.ucmerced.2i2c.cloud/ What do you think?

**Review by consideRatio — APPROVED** [review]
I agree with the adding a staging hub as a way to help the many users of ucmerced test the experience of choosing one of these images while migrating away from the one we have maintained.

Can we add a note that this is to be considered an exception, and that once they want the same experience in the prod hub, we remove this staging hub?

**Comment by sgibson91** [comment]
@GeorgianaElena what was the motivation for only using profile lists on staging? I actually don't see this mentioned in the original issue

**Comment by consideRatio** [comment]
@sgibson91 as a strategy to help them move away from using the old 2i2c maintained image, this relates to https://github.com/2i2c-org/infrastructure/issues/2674 and https://github.com/2i2c-org/infrastructure/issues/2583

**Comment by sgibson91** [comment]
Right, I just didn't think that was why the original issue was opened. I think it was opened because I tanked prod while trying to debug something right before a class 😅

**Comment by GeorgianaElena** [comment]
Sorry for not linking the main issue @sgibson91 (https://github.com/2i2c-org/infrastructure/issues/3188)

I have now also added a comment in `cluster.yaml` about this as well.

**Comment by consideRatio** [comment]
Oops I understood this initially as a "let this hub be deployed first to catch issues via our automation", but its more like a production hub that is deployed side by side with the real production ucmerced hub - used as a staging hub for the end users.

image

**Comment by sgibson91** [comment]
I'm surprised it didn't launch as a staging job given this logic

https://github.com/2i2c-org/infrastructure/blob/e9a7ac7802c5bc8855767ae03c2954743ff974f4/deployer/helm_upgrade_decision.py#L315-L322

**Pull request merged**